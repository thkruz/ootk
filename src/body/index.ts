export { Celestial } from './Celestial';
export { Earth } from './Earth';
export { Moon } from './Moon';
export { Sun } from './Sun';
