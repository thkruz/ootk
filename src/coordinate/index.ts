export * from './Hill.js';
export * from './ClassicalElements.js';
export * from './EquinoctialElements.js';
export * from './FormatTle.js';
export * from './Geodetic.js';
export * from './ITRF.js';
export * from './J2000.js';
export * from './RelativeState.js';
export * from './RIC.js';
export * from './StateVector.js';
export * from './TEME.js';
export * from './Tle.js';
