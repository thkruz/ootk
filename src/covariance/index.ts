export * from './CovarianceSample.js';
export * from './StateCovariance.js';
