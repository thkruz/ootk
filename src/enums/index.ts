export { CatalogSource } from './CatalogSource.js';
export { CommLink } from './CommLink.js';
export { OrbitRegime } from './OrbitRegime.js';
export { Sgp4OpsMode } from './Sgp4OpsMode.js';
export { AngularDiameterMethod } from './AngularDiameterMethod.js';
export { AngularDistanceMethod } from './AngularDistanceMethod.js';
export { PassType } from './PassType.js';
