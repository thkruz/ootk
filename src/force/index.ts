export { EarthGravity } from './EarthGravity.js';
export { Force } from './Force.js';
export { ForceModel } from './ForceModel.js';
export { Gravity } from './Gravity.js';
export { SolarRadiationPressure } from './SolarRadiationPressure.js';
export { ThirdBodyGravity } from './ThirdBodyGravity.js';
export { Thrust } from './Thrust.js';

