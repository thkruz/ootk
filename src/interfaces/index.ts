export type { BaseObjectParams } from './BaseObjectParams.js';
export type { GroundPositionParams } from './GroundPositionParams.js';
export type { OptionsParams } from './OptionsParams.js';
export type { SatelliteParams } from './SatelliteParams.js';
export type { SensorParams } from './SensorParams.js';
export type { StarObjectParams } from './StarObjectParams.js';
export type { ClassicalElementsParams } from './ClassicalElementsParams.js';
export type { EquinoctialElementsParams } from './EquinoctialElementsParams.js';
export type { RfSensorParams } from './RfSensorParams.js';
