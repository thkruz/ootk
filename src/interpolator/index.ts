export { ChebyshevCoefficients } from './ChebyshevCoefficients.js';
export { ChebyshevInterpolator } from './ChebyshevInterpolator.js';
export { CubicSpline } from './CubicSpline.js';
export { CubicSplineInterpolator } from './CubicSplineInterpolator.js';
export { Interpolator } from './Interpolator.js';
export { LagrangeInterpolator } from './LagrangeInterpolator.js';
export { StateInterpolator } from './StateInterpolator.js';
export { VerletBlendInterpolator } from './VerletBlendInterpolator.js';

