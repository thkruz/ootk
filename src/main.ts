/**
 * @author @thkruz Theodore Kruczek
 * @description Orbital Object ToolKit (ootk) is a collection of tools for working
 * with satellites and other orbital objects.
 * @license AGPL-3.0-or-later
 * @copyright (c) 2020-2024 Theodore Kruczek
 *
 * Many of the classes are based off of the work of @david-rc-dayton and his
 * Pious Squid library (https://github.com/david-rc-dayton/pious_squid).
 *
 * Orbital Object ToolKit is free software: you can redistribute it and/or modify it under the
 * terms of the GNU Affero General Public License as published by the Free Software
 * Foundation, either version 3 of the License, or (at your option) any later version.
 *
 * Orbital Object ToolKit is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License along with
 * Orbital Object ToolKit. If not, see <http://www.gnu.org/licenses/>.
 */

export * from './enums/index.js';
export * from './types/types.js';
export * from './interfaces/index.js';
export * from './time/index.js';
export * from './transforms/index.js';
export * from './utils/index.js';
export * from './operations/operations.js';
export { BaseObject, Sensor } from './objects/index.js';
export { Earth } from './body/index.js';
export * from './coordinate/index.js';
export * from './observation/index.js';
export * from './data/DataHandler.js';
export * from './sgp4/index.js';
export * from './objects/index.js';
export * from './interpolator/index.js';
export * from './force/index.js';
export * from './maneuver/index.js';
export * from './optimize/index.js';
export * from './orbit_determination/index.js';
export * from './body/index.js';
export * from './enums/index.js';
export * from './types/types.js';
export * from './interfaces/index.js';
export * from './coordinate/index.js';
export * from './operations/index.js';
export * from './force/index.js';
export * from './propagator/index.js';
export * from './transforms/index.js';
export * from './utils/index.js';
