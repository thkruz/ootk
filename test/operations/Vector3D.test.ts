import { Kilometers, KilometersPerSecond, Vector, Vector3D } from '../../src/main';

describe('Vector3D', () => {
  // fromVector
  it('should create a Vector3D from a Vector3D', () => {
    const v = Vector3D.fromVector(new Vector([1, 2, 3]));

    expect(v).toMatchSnapshot();
  });

  // toArray
  it('should return an array of elements', () => {
    const v = new Vector3D(1, 2, 3);

    expect(v.toArray()).toMatchSnapshot();
  });

  // getElement
  it('should get an element from a Vector3D', () => {
    const v = new Vector3D(1, 2, 3);

    expect(v.getElement(0)).toBe(1);
    expect(v.getElement(1)).toBe(2);
    expect(v.getElement(2)).toBe(3);
  });

  // toVector
  it('should return a Vector3D as a Vector', () => {
    const v = new Vector3D(1, 2, 3);

    expect(v.toVector()).toMatchSnapshot();
  });

  // toString
  it('should return a Vector3D as a string', () => {
    const v = new Vector3D(1, 2, 3);

    expect(v.toString()).toMatchSnapshot();
  });

  // distance
  it('should return the distance between two Vector3Ds', () => {
    const v1 = new Vector3D(1, 2, 3);
    const v2 = new Vector3D(4, 5, 6);

    expect((v1 as Vector3D).distance(v2)).toMatchSnapshot();
  });

  // outer
  it('should return the outer product of two Vector3Ds', () => {
    const v1 = new Vector3D(1, 2, 3);
    const v2 = new Vector3D(4, 5, 6);

    expect(v1.outer(v2)).toMatchSnapshot();
  });

  // skewSymmetric
  it('should return the skew symmetric of a Vector3D', () => {
    const v = new Vector3D(1, 2, 3);

    expect(v.skewSymmetric()).toMatchSnapshot();
  });

  // angleDegrees
  it('should return the angle between two Vector3Ds in degrees', () => {
    const v1 = new Vector3D<number>(1, 2, 3);
    const v2 = new Vector3D(4, 5, 6);

    expect(v1.angleDegrees(v2)).toMatchSnapshot();
  });

  // sight
  it('should return the sight of a Vector3D', () => {
    const v = new Vector3D(1, 2, 3);
    const v2 = new Vector3D(4 as KilometersPerSecond, 5 as KilometersPerSecond, 6 as KilometersPerSecond);

    expect(v.sight(v2, 20 as Kilometers)).toMatchSnapshot();
  });

  // bisect
  it('should return the bisect of two Vector3Ds', () => {
    const v1 = new Vector3D(1, 2, 3);
    const v2 = new Vector3D(4, 5, 6);

    expect((v1 as Vector3D).bisect(v2)).toMatchSnapshot();
  });

  // row
  it('should return the row of a Vector3D', () => {
    const v = new Vector3D(1, 2, 3);

    expect(v.row()).toMatchSnapshot();
  });

  // column
  it('should return the column of a Vector3D', () => {
    const v = new Vector3D(1, 2, 3);

    expect(v.column()).toMatchSnapshot();
  });

  // join
  it('should return the join of two Vector3Ds', () => {
    const v1 = new Vector3D(1, 2, 3);
    const v2 = new Vector3D(4, 5, 6);

    expect(v1.join(v2)).toMatchSnapshot();
  });
});
