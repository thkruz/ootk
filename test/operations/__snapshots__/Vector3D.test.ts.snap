// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vector3D should create a Vector3D from a Vector3D 1`] = `
Vector3D {
  "x": 1,
  "y": 2,
  "z": 3,
}
`;

exports[`Vector3D should return a Vector3D as a Vector 1`] = `
Vector {
  "elements": Array [
    1,
    2,
    3,
  ],
  "length": 3,
}
`;

exports[`Vector3D should return a Vector3D as a string 1`] = `"[1, 2, 3]"`;

exports[`Vector3D should return an array of elements 1`] = `
Float64Array [
  1,
  2,
  3,
]
`;

exports[`Vector3D should return the angle between two Vector3Ds in degrees 1`] = `12.93315449189912`;

exports[`Vector3D should return the bisect of two Vector3Ds 1`] = `
Vector3D {
  "x": 0.36386679548865963,
  "y": 0.5556981892812022,
  "z": 0.7475295830737448,
}
`;

exports[`Vector3D should return the column of a Vector3D 1`] = `
Matrix {
  "columns": 1,
  "elements": Array [
    Array [
      1,
    ],
    Array [
      2,
    ],
    Array [
      3,
    ],
  ],
  "rows": 3,
}
`;

exports[`Vector3D should return the distance between two Vector3Ds 1`] = `5.196152422706632`;

exports[`Vector3D should return the join of two Vector3Ds 1`] = `
Vector {
  "elements": Float64Array [
    1,
    2,
    3,
    4,
    5,
    6,
  ],
  "length": 6,
}
`;

exports[`Vector3D should return the outer product of two Vector3Ds 1`] = `
Matrix {
  "columns": 3,
  "elements": Array [
    Array [
      4,
      5,
      6,
    ],
    Array [
      8,
      10,
      12,
    ],
    Array [
      12,
      15,
      18,
    ],
  ],
  "rows": 3,
}
`;

exports[`Vector3D should return the row of a Vector3D 1`] = `
Matrix {
  "columns": 3,
  "elements": Array [
    Array [
      1,
      2,
      3,
    ],
  ],
  "rows": 1,
}
`;

exports[`Vector3D should return the sight of a Vector3D 1`] = `true`;

exports[`Vector3D should return the skew symmetric of a Vector3D 1`] = `
Matrix {
  "columns": 3,
  "elements": Array [
    Array [
      0,
      -3,
      2,
    ],
    Array [
      3,
      0,
      -1,
    ],
    Array [
      -2,
      1,
      0,
    ],
  ],
  "rows": 3,
}
`;
